client/components/sidebar.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useEffect } from 'react';
+
+export const Sidebar = () => {
+
+  const [ courseID, setCourseID ] = useState(10);
+  const [ priceData, setPriceData ] = useState();
+  const [ previewVideoData, setPreviewVideoData ] = useState();
+  const [ sidebarData, setSidebarData ] = useState();
+
+  useEffect(() => {
+    const controller = new AbortController();
+    const { signal } = controller;
+    let mounted = true;
+
+    fetch('http://localhost:3004/price?courseID=' + courseID, { signal })
+    .then(response => response.json())
+    .then(data => {
+      console.log("mounted: " + mounted);
+      if (mounted) {
+        setPriceData(data);
+      }
+    })
+    .catch(error => console.warn("Error: " + error.message));
+
+    return () => {
+      mounted = false;
+      controller.abort();
+    }
+
+  }, []);
+
+  useEffect(() => {
+    console.log(priceData !== undefined ? priceData.basePrice : "whoops");
+  });
+
+  let basePrice;
+  let discountPercentage;
+  let discountedPrice;
+  let saleEndDate;
+  let saleOngoing;
+
+
+  if (priceData !== undefined) {
+    basePrice = priceData.basePrice;
+    discountPercentage = priceData.discountPercentage;
+    discountedPrice = priceData.discountedPrice;
+    saleEndDate = priceData.saleEndDate;
+    saleOngoing = priceData.saleOngoing;
+  }
+
+  return (
```

Comment:
For readability purposes, it may be beneficial to refactor into more sub-components, and split up some of the logic. I see that a lot of the divs don't have inner HTML so I'm guessing it's on your todos for later

Reply:
Yeah; I was thinking about that. Right, there's still a lot of HTML to add here, so I'll see how big this gets and split off some components if that seems like a good idea.